components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
 
+subroutine phys_run2(phys_state, ztodt, phys_tend, pbuf2d, cam_out, cam_in )
+  !-----------------------------------------------------------------------------
+  ! Purpose: Second part of atmos physics after updating of surface models
+  !-----------------------------------------------------------------------------
+  use physics_buffer,   only: physics_buffer_desc, pbuf_get_chunk, pbuf_deallocate, pbuf_update_tim_idx
+  use cam_diagnostics,  only: diag_deallocate, diag_surf
+  use comsrf,           only: trefmxav, trefmnav, sgh, sgh30, fsds 
+  use physconst,        only: stebol, latvap
+  use time_manager,     only: get_nstep
+  use check_energy,     only: ieflx_gmean, check_ieflx_fix 
+  use phys_control,     only: ieflx_opt
+  !-----------------------------------------------------------------------------
+  ! Interface arguments
+  !-----------------------------------------------------------------------------
+  real(r8), intent(in) :: ztodt                       ! physics time step unless nstep=0
+  type(physics_state), intent(inout), dimension(begchunk:endchunk) :: phys_state
+  type(physics_tend ), intent(inout), dimension(begchunk:endchunk) :: phys_tend
+  type(physics_buffer_desc), pointer, dimension(:,:)               :: pbuf2d
+  type(cam_out_t),     intent(inout), dimension(begchunk:endchunk) :: cam_out
+  type(cam_in_t),      intent(inout), dimension(begchunk:endchunk) :: cam_in
+  !-----------------------------------------------------------------------------
+  ! Local Variables
+  !-----------------------------------------------------------------------------
+  integer :: c                                 ! chunk index
+  integer :: ncol                              ! number of columns
+  integer :: nstep                             ! current timestep number
+  !-----------------------------------------------------------------------------
+  !-----------------------------------------------------------------------------
 #if (! defined SPMD)
-    integer :: mpicom = 0
-#endif
-    integer(i8) :: beg_count, end_count, irtc_rate ! for measuring chunk cost
-    real(r8):: chunk_cost
-    type(physics_buffer_desc),pointer, dimension(:)     :: phys_buffer_chunk
-    !
-    ! If exit condition just return
-    !
-
-    if(single_column.and.scm_crm_mode) return
-
-    if ( adiabatic .or. ideal_phys ) return
-    !-----------------------------------------------------------------------
-    ! Tendency physics after coupler 
-    ! Not necessary at terminal timestep.
-    !-----------------------------------------------------------------------
-    !
-#if ( defined OFFLINE_DYN )
-    !
-    ! if offline mode set SHFLX QFLX TAUX TAUY for vert diffusion
-    !
-    call get_met_srf2( cam_in )
+  integer :: mpicom = 0
 #endif
-    ! Set lightning production of NO
-    call t_startf ('lightning_no_prod')
-    call lightning_no_prod( phys_state, pbuf2d,  cam_in )
```

Comment:
We lost lightning production of NO. Is this intentional?

Reply:
oh... no, I should put that back

Reply:
restored